config/src/main/java/com/typesafe/config/impl/SimpleConfig.java

```diff
@@ -884,7 +902,48 @@ public static long parseBytes(String input, ConfigOrigin originForException,
                 return result.longValue();
             else
                 throw new ConfigException.BadValue(originForException, pathForException,
-                        "size-in-bytes value is out of range for a 64-bit long: '" + input + "'");
+                    "size-in-bytes value is out of range for a 64-bit long: '" + input + "'");
+        } catch (NumberFormatException e) {
+            throw new ConfigException.BadValue(originForException, pathForException,
+                "Could not parse size-in-bytes number '" + numberString + "'");
+        }
+    }
+
+
+    public static BigInteger parseBytesAsBigInteger(String input, ConfigOrigin originForException,
```

Comment:
implementing the Long version in terms of this one (with added range checking) would make sense to me. I think the code duplication isn't worth it to avoid allocating a BigInteger

Reply:
yep, thank you! fixed.